components/ReviewTransactionModal.tsx

```diff
@@ -0,0 +1,154 @@
+  const network = NetworkName.EthereumGoerli; // TODO: Remove once we support multiple chains
+  const { getShieldPrivateKey } = useShieldPrivateKey();
+  const { isConnected, address } = useAccount();
+  const provider = useProvider();
+  const { data: signer } = useSigner();
+  const { txNotify } = useNotifications();
+  const { shield, isShielding } = useRailgunTx();
+  const bigNumberAmount = parseUnits(tokenAmount! || "0", tokenDecimals);
+  const feeAmount = parseUnits(tokenAmount! || "0", tokenDecimals).div("400");
+  const total = feeAmount.add(bigNumberAmount);
+
+  const doSubmit: React.FormEventHandler = async (e) => {
+    // TODO: Form validation
+    if (!tokenAddress || !tokenAmount || !tokenDecimals || !recipient)
+      throw new Error("bad form");
+    const tx = await shield({
+      tokenAddress: tokenAddress,
+      tokenAmount: tokenAmount,
+      tokenDecimals: tokenDecimals,
+      recipient: recipient,
+    });
+    txNotify(tx!.hash);
+    onClose();
+  };
+
+  return (
+    <Modal onClose={onClose} isOpen={isOpen} isCentered>
+      <ModalOverlay />
+      <ModalContent>
+        <ModalHeader>Confirm Shield</ModalHeader>
+        <ModalCloseButton />
+        <ModalBody>
+          <Flex direction="column">
+            <Flex
+              direction="column"
+              borderRadius="1.5rem"
+              border="1px solid black"
+              padding="1rem"
+            >
+              <Flex align="center" justify="space-between">
+                <Heading size="xs" paddingX={2}>
+                  Recipient
+                </Heading>
+                <Tooltip label={recipient}>
+                  <Text size="sm" textOverflow="ellipsis">
+                    {shortenAddress(recipient)}
+                  </Text>
+                </Tooltip>
+              </Flex>
+              <Flex align="center" justify="space-between">
+                <Heading size="xs" paddingX={2}>
+                  Token name
+                </Heading>
+                <Text size="sm">{tokenName}</Text>
+              </Flex>
+              <Flex align="center" justify="space-between">
+                <Heading size="xs" paddingX={2}>
+                  Amount
+                </Heading>
+                <Text size="sm">
+                  {ethers.utils.formatUnits(bigNumberAmount, tokenDecimals)}{" "}
+                  {tokenSymbol}
+                </Text>
+              </Flex>
+              <Flex align="center" justify="space-between">
+                <Heading size="xs" paddingX={2}>
+                  Shielding fee
+                </Heading>
+                <Text size="sm">
+                  {ethers.utils.formatUnits(feeAmount, tokenDecimals)}{" "}
+                  {tokenSymbol}
+                </Text>
+              </Flex>
+            </Flex>
+            <Flex align="center" justify="space-between" marginTop="1.5rem">
+              <Heading size="sm" paddingX={2}>
+                Amount + Fee
```

Comment:
I believe the way this works is that fee is taken out of amount. In other words, if you send 1 ETH, and the fee % is 5%, then actually, you'll be sending .95 ETH, with the .05 ETH fee being deducted.

Reply:
Ah okay, didn't know that, I will remove this Amount + fee line item from the modal

Reply:
This has been removed